com.zsmartsystems.zigbee/src/main/java/com/zsmartsystems/zigbee/app/otaserver/ZclOtaUpgradeServer.java

```diff
@@ -514,40 +524,55 @@ public void run() {
     /**
      * Sends an image block to the client
      *
+     * @param command the request command. May be null.
      * @param fileOffset the offset into the {@link ZigBeeOtaFile} to send the block
      * @param maximumDataSize the maximum data size the client can accept
      * @return the number of bytes sent
      */
-    private int sendImageBlock(int fileOffset, int maximumDataSize) {
+    private int sendImageBlock(ZclCommand command, int fileOffset, int maximumDataSize) {
         ByteArray imageData = otaFile.getImageData(fileOffset, Math.min(dataSize, maximumDataSize));
         logger.debug("{} OTA Data: Sending {} bytes at offset {}", cluster.getZigBeeAddress(), imageData.size(),
                 fileOffset);
-        cluster.imageBlockResponse(ZclStatus.SUCCESS, otaFile.getManufacturerCode(), otaFile.getImageType(),
-                otaFile.getFileVersion(), fileOffset, imageData);
+
+        if (command == null) {
+            cluster.sendCommand(new ImageBlockResponse()
```

Comment:
Can we extract
```
new ImageBlockResponse()
                    .setStatus(ZclStatus.SUCCESS)
                    .setManufacturerCode(otaFile.getManufacturerCode())
                    .setImageType(otaFile.getImageType())
                    .setFileVersion(otaFile.getFileVersion())
                    .setFileOffset(fileOffset)
                    .setImageData(imageData)
```
to a variable before the if and then within if/else only call setDisableDefaultResponse and use proper sending method? Or get the setDisableDefaultResponse outside the if and call it with `cluster==null` arg
o

Reply:
Yes, Good point